blog/2021-09-17-announcing-stryker-net-1.md

```diff
@@ -0,0 +1,265 @@
+"stryker-config": {
+    "ignore-mutations": [
+        "linq.First",
+        "linq.Sum"
+    ]
+}
+```
+
+> Note: this only works for Linq mutations for now.
+
+All possible mutations:
+
+```
+None,
+Distinct,
+Reverse,
+Average,
+AsEnumerable,
+OrderBy,
+OrderByDescending,
+FirstOrDefault,
+First,
+SingleOrDefault,
+Single,
+Last,
+All,
+Any,
+Skip,
+Take,
+SkipWhile,
+TakeWhile,
+Min,
+Max,
+Sum,
+Count,
+ThenBy,
+ThenByDescending,
+Union,
+Intersect,
+Concat,
+Except
+```
+
+### MsBuild path option
+
+By default stryker tries to autodiscover msbuild on your system. If stryker fails to discover msbuild you may supply the path to msbuild manually with this option.
+
+Example:
+
+```shell
+--msbuild-path "c://MsBuild/MsBuild.exe"
+```
+
+### Target framework
+
+If the project targets multiple frameworks, it is now possible to specify the particular framework to build against. If you specify a non-existent target, Stryker will build the project against a random one (or the only one if so).
+
+Example:
+
+```json
+{
+  "stryker-config": {
+    "target-framework": "netcoreapp3.1"
+  }
+}
+```
+
+### Filter test cases
+
+A long awaited feature has finally found its way into Stryker! It is now possible to exclude some test cases. For example if you have long running integration tests in your unit test project they can be disabled for Stryker. This will improve runtime.
+
+Example:
+
+```json
+{
+  "stryker-config": {
+    "test-case-filter": "(FullyQualifiedName~UnitTest1&TestCategory=CategoryA)|Priority=1"
+  }
+}
+```
+
+Uses `dotnet test --filter` option syntax, [detailed here](https://docs.microsoft.com/en-us/dotnet/core/testing/selective-unit-tests).
+
+### Use source link for dashboard reporter
+
+Filling all settings to use the dashboard reporter could be a bit of a hustle. Thanks to [SourceLink](https://github.com/dotnet/sourcelink#readme), the repository URL and the full version (including the git SHA1) of a project can be included in the produced assembly.
+
+Stryker now uses the information computed by SourceLink to automatically retrieve the project name (github.com/organization/project) and project version which are required for the dashboard reporter.
+
+Enable this by adding the following to your `.csproj`:
+
+```xml
+  <ItemGroup>
+    <PackageReference Include="DotNet.ReproducibleBuilds" Version="0.1.66" PrivateAssets="All"/>
+  </ItemGroup>
+```
+
+For full info on how to enable SourceLink using ReproducibleBuilds see [their readme](https://github.com/dotnet/reproducible-builds/blob/main/README.md)
+
+## 🐛 Bug fixes
+
+### No more mutated assembly on disk after stryker run
+
+Internally Stryker replaces the `.dll` on your disk when running. Up till now the mutated assembly stayed in place. This had some unintended side effects. For example code coverage results are incorrect when run on a mutated assembly. And there was the risk of accidentally releasing/publishing the mutated assembly instead of the original one in a pipeline.
```

Comment:
```suggestion
While mutation testing Stryker replaces your system under test assembly on disk. Up till now the mutated assembly stayed in place after mutation testing was done. This had some unintended side effects. For example code coverage results could be incorrect until you rebuild your project and there was the risk of accidentally releasing/publishing the mutated assembly instead of the original if you did not rebuild you project after mutation testing in your pipelines. We now copy your original assembly before we modify it, and place it back after we're done. No more rebuild required!
```